aws/logs_monitoring/lambda_function.py

```diff
@@ -132,50 +138,112 @@ def __exit__(self, ex_type, ex_value, traceback):
             print("DatadogConnection exit: ", ex_type, ex_value, traceback)
 
 
+class DatadogHTTPClient(object):
+
+    _POST = "POST"
+    _HEADERS = {"Content-type": "application/json"}
+
+    def __init__(self, host, apiKey, timeout=1, max_retries=1, backoff=0.5):
+        self._url = "https://{}/v1/input/{}".format(host, apiKey)
+        self._session = requests.Session()
+        self._session.headers.update(self._HEADERS)
+        self._timeout = timeout if timeout > 0 else 1
+        self._max_retries = max_retries if max_retries >= 0 else 1
+        self._backoff = backoff if backoff > 0 else 0.5
```

Comment:
optionally you could use what is proposed here https://www.peterbe.com/plog/best-practice-with-retries-with-requests